src/test/java/NestedNullTesting.java

```diff
@@ -0,0 +1,81 @@
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import org.apache.commons.io.IOUtils;
+import org.checkerframework.com.google.common.collect.Lists;
+import org.eclipse.jdt.core.dom.AST;
+import org.eclipse.jdt.core.dom.ASTParser;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.core.dom.Expression;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+/**
+ * Class to perform JUnit tests on the NestedNullRefactoring refactoring module
+ * 
+ * @see NestedNullRefactoring
+ */
+public class NestedNullTesting {

```

Comment:
These tests are not the ones that the PR says should pass. Are they expected to fail? In general, I am hesitant to merge any PR for which the tests are failing - the code on the `main` branch should always be kept in a good state.